server/controllers/stripe.js

```diff
@@ -45,29 +35,51 @@ exports.retrieveCustomer = async (req, res) => {
   res.send(customer);
 };
 
-// @route POST /stripe/payment/create
-// @desc creates new payment method
+// @route GET /stripe/customers/retrieveAll
+// @desc retrieve customer
 // @access Public
-exports.createPaymentMethod = async (req, res) => {
-  const paymentMethod = await stripe.paymentMethods.create({
-    type: "card",
-    card: {
-      number: "4242424242424242",
-      exp_month: 12,
-      exp_year: 2022,
-      cvc: "314",
-    },
-  });
-  res.send(paymentMethod);
+exports.retrieveAllCustomers = async (req, res) => {
+  const customers = await stripe.customers.list();
+  res.send(customers);
 };
 
-// @route GET /stripe/payment/all
+// @route GET /stripe/payment/all/:customerId
 // @desc gets all payment methods
-// @access Public
+// @access Private
 exports.getAllPaymentMethods = async (req, res) => {
   const paymentMethods = await stripe.paymentMethods.list({
-    customer: "cus_Kk1rd4KHOzlRC0",
+    customer: req.user.stripeId,
```

Comment:
When we perform the authentication in our middleware, we need to add the `stripeId` to the request object as well, I don't think we are currently doing that!

Please see: `/server/middleware/auth.js`

Reply:
I couldn't figure out how to change the auth middleware, as I am unfamiliar with jwt. Instead, I added in a call to find the user via req.user.id, and then used user.stripeId inside the stripe call